modules/core/src/mathfuncs.cpp

```diff
@@ -1574,7 +1574,7 @@ static bool ocl_patchNaNs( InputOutputArray _a, float value )
     int rowsPerWI = ocl::Device::getDefault().isIntel() ? 4 : 1;
     ocl::Kernel k("KF", ocl::core::arithm_oclsrc,
                      format("-D UNARY_OP -D OP_PATCH_NANS -D dstT=float -D DEPTH_dst=%d -D rowsPerWI=%d",
-                            CV_32F, rowsPerWI));
+                            _a.depth(), rowsPerWI));
```

Comment:
Extra check is required for OpenCL devices without doubles (64F) support.

Look for code with:

```
bool doubleSupport = d.doubleFPConfig() > 0;
```